### PR TITLE
Allow lazy fragment initializaiton through promise

### LINKED ISSUE #92: Ability to consume promises from init function of fragments
This is helpful if the fragment wants to render lazily or asynchronously.

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ This is a backward compatible change

fixes #92

### CODE REVIEW

File: src/pipe.js [hunk 1]
```diff
@@ -46,21 +46,48 @@ var Pipe = (function (doc, perf) { //eslint-disable-line no-unused-vars, strict
...
+                    perf.clearMarks(fragmentId);
+                    perf.clearMarks(fragmentId + '-end');
+                }
+
+                (function executeAndCapture() {
+                    isUTSupported && perf.mark(fragmentId);
+                    // Check if the exposed function is a Promise to allow lazy rendering
+                    // Capture initializaion cost of each fragment on the page using User Timing API if available
+                    if (isPromise(init)) {
+                        init.then(function(module) {
```
Comment (grassator) [inline src/pipe.js]: I think that `init` function should rather return a promise

File: src/pipe.js [hunk 2]
```diff
@@ -46,21 +46,46 @@ var Pipe = (function (doc, perf) { //eslint-disable-line no-unused-vars, strict
...
-                        perf.mark(fragmentId);
-                        init(node);
+                // early return
+                if (typeof init !== 'function') {
+                    return;
+                }
+                // check if User Timing API is supported
+                var isUTSupported = perf && 'mark' in perf;
+
+                function isPromise(obj) {
```
Comment (dazld) [inline src/pipe.js]: pull this out into a library function and import it

Comment (vigneshshanmugam) [inline src/pipe.js]: No.. then I would need a built tool and I dont want to go in that path.

Comment (dazld) [inline src/pipe.js]: maybe, but it's probably something that should be covered by a test, so pulling it out into a library func is going to help with that.

Comment (vigneshshanmugam) [inline src/pipe.js]: Yeah i didnt do that since its not used anywhere other than this one place..

Comment (dazld) [inline src/pipe.js]: oki

**Pull request merged**